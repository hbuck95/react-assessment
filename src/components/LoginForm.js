import React, { Component } from 'react';
import axios from 'axios';

import {
    Button,
    Container,
    Row,
    Col
} from 'reactstrap';

import 'bootstrap/dist/css/bootstrap.css'

export default class LoginForm extends Component {
    constructor() {
        super();
        this.state = {
            loginStatus: null
        };
    }

    handleSubmit = (e) => {
        e.preventDefault();

        axios.get(`http://localhost:5000/user/name/${e.target[0].value}/${e.target[1].value}`)
            .then(response => {
                if (response.data.Status === "Logged In") {
                    document.getElementById("loginForm").reset();
                    this.setState({ loginStatus: "Login successful!" });
                } else {
                    this.setState({ loginStatus: "Incorrect username/password combination." });
                }
            })
            .catch(err => {
                console.log(err);
            });
    }

    render() {
        return (
            <div id="register">
                <Container>
                    <br />
                    <form id="loginForm" onSubmit={this.handleSubmit}>
                        <Row>
                            <Col />
                            <Col />
                            <Col />
                            <Col className="formHead" xs="auto"><h2>Please enter your account details:</h2></Col>
                            <Col />
                            <Col />
                            <Col />
                        </Row>
                        <Row>
                            <Col />
                            <Col />
                            <Col><h5 className="formField">Username:</h5></Col>
                            <Col><input id="usernameInput" size="25" type="text" placeholder="bobgog123"></input></Col>
                            <Col />
                            <Col />
                            <Col />
                        </Row>
                        <Row>
                            <Col />
                            <Col />
                            <Col><h5 className="formField">Password:</h5></Col>
                            <Col><input id="passwordInputA" size="25" type="password" placeholder="Must not be empty"></input></Col>
                            <Col />
                            <Col />
                            <Col />
                        </Row>
                        <br />
                        <Row>
                            <Col />
                            <Col />
                            <Col><Button id="userSubmit" color={this.state.loginStatus === null ? "primary" : this.state.loginStatus === "Login successful!" ? "success" : "danger" }>Login</Button></Col>
                            <Col />
                            <Col />
                        </Row>
                        <br />
                        <Row>
                            <Col />
                            <Col />
                            <Col />
                            <Col className="errorField"><h5>{this.state.loginStatus === null ? "" : this.state.loginStatus}</h5></Col>
                            <Col />
                            <Col />
                            <Col />
                        </Row>
                    </form>
                </Container>
            </div>
        );
    }
}