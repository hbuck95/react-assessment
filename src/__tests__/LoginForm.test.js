import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';

import LoginForm from '../components/LoginForm';

configure({adapter: new Adapter()});

it('LoginForm renders', () => {
    const compJson = renderer.create(<LoginForm/>).toJSON();
    expect(compJson).toMatchSnapshot();
  });
