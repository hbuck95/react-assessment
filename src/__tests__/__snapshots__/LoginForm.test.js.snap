// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders 1`] = `
<div
  id="register"
>
  <div
    className="container"
  >
    <br />
    <form
      id="loginForm"
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="formHead col-auto"
        >
          <h2>
            Please enter your account details:
          </h2>
        </div>
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        >
          <h5
            className="formField"
          >
            Username:
          </h5>
        </div>
        <div
          className="col"
        >
          <input
            id="usernameInput"
            placeholder="bobgog123"
            size="25"
            type="text"
          />
        </div>
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        >
          <h5
            className="formField"
          >
            Password:
          </h5>
        </div>
        <div
          className="col"
        >
          <input
            id="passwordInputA"
            placeholder="Must not be empty"
            size="25"
            type="password"
          />
        </div>
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        />
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        >
          <button
            aria-label={null}
            className="btn btn-primary"
            id="userSubmit"
            onClick={[Function]}
          >
            Login
          </button>
        </div>
        <div
          className="col"
        />
        <div
          className="col"
        />
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="errorField col"
        >
          <h5>
            
          </h5>
        </div>
        <div
          className="col"
        />
        <div
          className="col"
        />
        <div
          className="col"
        />
      </div>
    </form>
  </div>
</div>
`;
